nvflare/private/fed/client/client_runner.py

```diff
@@ -342,75 +339,77 @@ def _process_task(self, task: TaskAssignment, fl_ctx: FLContext) -> Shareable:
-        task_fetch_interval = self.task_fetch_interval
         while not self.asked_to_stop:
             with self.engine.new_context() as fl_ctx:
                 if self.run_abort_signal.triggered:
                     self.log_info(fl_ctx, "run abort signal received")
                     break
 
-                time.sleep(task_fetch_interval)
+                task_fetch_interval, _ = self.fetch_and_run_one_task(fl_ctx)
 
                 if self.run_abort_signal.triggered:
                     self.log_info(fl_ctx, "run abort signal received")
                     break
 
-                task_fetch_interval, _ = self.run_one_task(fl_ctx)
+                time.sleep(task_fetch_interval)
+
+    def fetch_and_run_one_task(self, fl_ctx) -> (float, bool):
+        """Fetches and runs a task.
 
-    def run_one_task(self, fl_ctx):
-        # reset to default fetch interval
-        task_fetch_interval = self.task_fetch_interval
+        Returns:
+            A tuple of (task_fetch_interval, task_processed).
+        """
+        default_task_fetch_interval = 0.1
         self.log_debug(fl_ctx, "fetching task from server ...")
         task = self.engine.get_task_assignment(fl_ctx)
-        task_processed = False
-        if not task:
-            self.log_debug(fl_ctx, "no task received - will try in {} secs".format(task_fetch_interval))
-            # continue
 
+        if not task:
+            self.log_debug(fl_ctx, "no task received - will try in {} secs".format(default_task_fetch_interval))
+            return default_task_fetch_interval, False
         elif task.name == SpecialTaskName.END_RUN:
             self.log_info(fl_ctx, "server asked to end the run")
-            # break
             self.asked_to_stop = True
-
+            return default_task_fetch_interval, False
         elif task.name == SpecialTaskName.TRY_AGAIN:
             task_data = task.data
+            task_fetch_interval = default_task_fetch_interval
             if task_data and isinstance(task_data, Shareable):
-                task_fetch_interval = task_data.get(TaskConstant.WAIT_TIME, self.task_fetch_interval)
+                task_fetch_interval = task_data.get(TaskConstant.WAIT_TIME, task_fetch_interval)
             self.log_debug(fl_ctx, "server asked to try again - will try in {} secs".format(task_fetch_interval))
-            # continue
+            return task_fetch_interval, False
 
-        else:
-            self.log_info(fl_ctx, "got task assignment: name={}, id={}".format(task.name, task.task_id))
+        self.log_info(fl_ctx, "got task assignment: name={}, id={}".format(task.name, task.task_id))
 
-            # create a new task abort signal
-            task_reply = self._process_task(task, fl_ctx)
+        task_data = task.data
+        if not isinstance(task_data, Shareable):
+            raise TypeError("task_data must be Shareable, but got {}".format(type(task_data)))
+        task_fetch_interval = task_data[TaskConstant.WAIT_TIME]
```

Comment:
Should be defensive here in case the task_data doesn't contain WAIT_TIME: task_fetch_interval = task_data.get(TaskConstant.WAIT_TIME, task_fetch_interval)

Reply:
thanks, done.